jib-maven-plugin/src/main/java/com/google/cloud/tools/jib/maven/JibPluginConfiguration.java

```diff
@@ -417,6 +458,37 @@ String getFormat() {
     return Preconditions.checkNotNull(container.format);
   }
 
+  /**
+   * Gets the configured extra directory path.
+   *
+   * @return the configured extra directory path
+   */
+  @Nullable
+  Path getExtraDirectoryPath() {
+    // TODO: Should inform user about nonexistent directory if using custom directory.
+    if (System.getProperty(PropertyNames.EXTRA_DIRECTORY_PATH) != null) {
+      return Paths.get(System.getProperty(PropertyNames.EXTRA_DIRECTORY_PATH));
+    }
+    return extraDirectory.path == null ? null : extraDirectory.path.toPath();
+  }
+
+  /**
+   * Gets the configured extra layer file permissions.
+   *
+   * @return the configured extra layer file permissions
+   */
+  List<PermissionConfiguration> getExtraDirectoryPermissions() {
+    if (System.getProperty(PropertyNames.EXTRA_DIRECTORY_PERMISSIONS) != null) {
```

Comment:
Do you really expect this to happen?

Reply:
Not particularly, but every other parameter has a corresponding system property, so I think it's best to keep it consistent.